### PR TITLE
DOCX lists [SCI-5043] 

### PR DESCRIPTION
**Pull request by urbanrotnik**
Jira ticket: [SCI-5043](https://biosistemika.atlassian.net/browse/SCI-5043)

### What was done
Magic was done with "predictable" caracal gem :) 

Supported the nested checklist with images and tables elements. 
Images and tables will be rendered after the checklist but the `li` item will have an internal link to them. 
caracal do not support tables, images inside of `li` elements.

### CODE REVIEW

File: app/services/reports/docx.rb [hunk 1]
```diff
@@ -81,6 +86,103 @@ def self.render_img_element(docx, element, options = {})
...
+        docx.bookmark_start id: index, name: key
+        # How to draw table here?
+        # docx = Caracal::Document
+        # self = Reports::Docx
+        # But you have instance method on Reports::Docx. How to access Reports::Docx of current docx?
+        # docx.tiny_mce_table(item)
+        docx.p do
+          text 'Table here soon'
+        end
+        docx.bookmark_end id: index
```
Comment (urbanrotnik) [inline app/services/reports/docx.rb:108]: @aignatov-bio Any ideas here?

File: app/services/reports/docx.rb [hunk 2]
```diff
@@ -81,6 +86,88 @@ def self.render_img_element(docx, element, options = {})
...
-end
 
+  def self.render_list_element(docx, element, options = {})
+    bookmark_items = Reports::Docx.recursive_list_items_renderer(docx, element)
+
+    bookmark_items.each_with_index do |(key, item), index|
+      if item[:type] == 'image'
+        docx.bookmark_start id: index, name: key
+        docx.p do
+          br
```
Comment (aignatov-bio) [inline app/services/reports/docx.rb:97]: Why do you add a new line here? I think the `p` element already create newline

File: app/services/reports/docx.rb [hunk 3]
```diff
@@ -81,6 +86,88 @@ def self.render_img_element(docx, element, options = {})
...
+          values_array.each do |item|
+            case item
+            when Hash
+              if %w(ul ol li).include?(item[:type])
+                Reports::Docx.recursive_list_items_renderer(self, item, bookmark_items: bookmark_items)
+              elsif %w(a).include?(item[:type])
+                Reports::Docx.render_link_element(self, item)
+              elsif %w(image).include?(item[:type])
+                bookmark_items[item[:bookmark_id]] = item
+                link "Appended image - #{item[:blob]&.filename}", item[:bookmark_id] do
```
Comment (aignatov-bio) [inline app/services/reports/docx.rb:130]: I18n?

File: app/services/reports/docx.rb [hunk 4]
```diff
@@ -81,6 +86,88 @@ def self.render_img_element(docx, element, options = {})
...
+                  internal true
+                end
+              elsif %w(table).include?(item[:type])
+                bookmark_items[item[:bookmark_id]] = item
+                link 'Appended table', item[:bookmark_id] do
```
Comment (aignatov-bio) [inline app/services/reports/docx.rb:135]: I18n?

**Review by aignatov-bio — CHANGES_REQUESTED** [review]
Maybe better to move all methods related to rendering tinymce to separate file.

File: app/services/reports/docx.rb (hunk 2 above)
Comment (urbanrotnik) [reply app/services/reports/docx.rb:97]: To have some top margin

File: app/services/reports/docx.rb (hunk 3 above)
Comment (urbanrotnik) [reply app/services/reports/docx.rb:130]: yup

File: app/services/reports/docx.rb (hunk 4 above)
Comment (urbanrotnik) [reply app/services/reports/docx.rb:135]: yup

**Review by aignatov-bio — CHANGES_REQUESTED** [review]
Fix smart annotation errors

**Review by aignatov-bio — APPROVED** [review]

**Pull request merged**